### PR TITLE
pacific: mgr/devicehealth: do not create pool until there are enough osds

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
retest this please

**Review by neha-ojha — APPROVED** [review]

**Comment by neha-ojha** [comment]
jenkins test api

File: src/pybind/mgr/devicehealth/module.py [hunk 1]
```diff
@@ -211,12 +211,29 @@ def config_notify(self):
...
+            self.maybe_create_device_pool()
+
+    def have_enough_osds(self) -> bool:
+        # wait until we have enough OSDs to allow the pool to be healthy
+        up = 0
+        for osd in self.get("osd_map")["osds"]:
+            if osd["up"]:
+                up += 1
+
+        need = cast(int, self.get_ceph_option("osd_pool_default_size"))
```
Comment (neha-ojha) [inline src/pybind/mgr/devicehealth/module.py:226]: pacific version of devicehealth does not use cast 

```
2021-02-06T21:18:02.301 INFO:tasks.ceph.mgr.x.smithi198.stderr:2021-02-06T21:18:02.301+0000 7fbbb1bfc700 -1 log_channel(cluster) log [ERR] : Unhandled exception from module 'devicehealth' while running on mgr.x: name 'cast' is not defined
2021-02-06T21:18:02.302 INFO:tasks.ceph.mgr.x.smithi198.stderr:2021-02-06T21:18:02.301+0000 7fbbb1bfc700 -1 devicehealth.serve:
2021-02-06T21:18:02.302 INFO:tasks.ceph.mgr.x.smithi198.stderr:2021-02-06T21:18:02.301+0000 7fbbb1bfc700 -1 Traceback (most recent call last):
2021-02-06T21:18:02.303 INFO:tasks.ceph.mgr.x.smithi198.stderr:  File "/usr/share/ceph/mgr/devicehealth/module.py", line 298, in serve
2021-02-06T21:18:02.303 INFO:tasks.ceph.mgr.x.smithi198.stderr:    self.scrape_all()
2021-02-06T21:18:02.303 INFO:tasks.ceph.mgr.x.smithi198.stderr:  File "/usr/share/ceph/mgr/devicehealth/module.py", line 337, in scrape_all
2021-02-06T21:18:02.303 INFO:tasks.ceph.mgr.x.smithi198.stderr:    ioctx = self.open_connection()
2021-02-06T21:18:02.304 INFO:tasks.ceph.mgr.x.smithi198.stderr:  File "/usr/share/ceph/mgr/devicehealth/module.py", line 316, in open_connection
2021-02-06T21:18:02.304 INFO:tasks.ceph.mgr.x.smithi198.stderr:    if not self.maybe_create_device_pool():
2021-02-06T21:18:02.304 INFO:tasks.ceph.mgr.x.smithi198.stderr:  File "/usr/share/ceph/mgr/devicehealth/module.py", line 231, in maybe_create_device_pool
2021-02-06T21:18:02.304 INFO:tasks.ceph.mgr.x.smithi198.stderr:    if not self.have_enough_osds():
2021-02-06T21:18:02.304 INFO:tasks.ceph.mgr.x.smithi198.stderr:  File "/usr/share/ceph/mgr/devicehealth/module.py", line 226, in have_enough_osds
2021-02-06T21:18:02.305 INFO:tasks.ceph.mgr.x.smithi198.stderr:    need = cast(int, self.get_ceph_option("osd_pool_default_size"))
2021-02-06T21:18:02.305 INFO:tasks.ceph.mgr.x.smithi198.stderr:NameError: name 'cast' is not defined
```

**Comment by yuriw** [comment]
@liewegas this failed tests https://pulpito.ceph.com/?branch=wip-yuri2-testing-2021-02-15-1007-pacific

**Comment by liewegas** [comment]
Updated the PR to use ``int(...)`` instead of ``cast(int, ...)``... @sebastian-philipp is that right?

**Comment by yuriw** [comment]
[wip-yuri7-testing-2021-02-22-0837-pacific](https://trello.com/c/ps8f45yv/1151-wip-yuri7-testing-2021-02-22-0837-pacific)

**Comment by liewegas** [comment]
jenkins test make check

**Comment by neha-ojha** [comment]
jenkins test make check

**Review by neha-ojha — APPROVED** [review]

**Pull request merged**